client/client-multi/client-c/src/kaa/platform-impl/common/sha.c

```diff
@@ -24,7 +24,7 @@
 
 kaa_error_t ext_calculate_sha_hash(const char *data, size_t data_size, kaa_digest digest)
 {
-    if (!digest || !data || !data_size) {
+    if (!digest || (!data && data_size)) {
```

Comment:
I agree with Maxim. 
Also, take a look at the _mbedtls_sha1_ function in mbedtls library. It does the same steps that are represented below. I'd recommend to use it.

Also, is it really necessary to call this function with NULL buffer?
I mean _ext_calculate_sha_hash(NULL, 0, profile_manager->profile_hash);_ in kaa_profile.c

What does it mean? Let me say, it is not the SHA-1 from zero.
Ideally,  I'd prefer to leave this function without changes and implement better logic into the caller function.